docs/src/ref/modeling.md

```diff
@@ -97,31 +97,46 @@ val::Bool = @trace(bernoulli(0.5), :z)
 Not all random choices need to be given addresses.
 An address is required if the random choice will be observed, or will be referenced by a custom inference algorithm (e.g. if it will be proposed to by a custom proposal distribution).
 
-### Choices should have constant support
-The support of a random choice at a given address (the set of values with nonzero probability or probability density) must be constant across all possible executions of the `@gen` function.
-Violating this discipline will cause errors in certain cases.
-If the support of a random choice needs to change, use a different address for each distinct value of the support.
-For example, consider the following generative function:
+
+### Sample space and support of random choices
+
+Different probability distributions produce different types of values for their random choices. For example, the [`Bernoulli`](@ref) distribution results in `Bool` values (either `true` or `false`), the [`Normal`](@ref) distribution results in `Real` values that may be positive or negative, and the [`Beta`](@ref) distributions result in `Real` values that are always in the unit interval (0, 1).
+
+Each [`Distribution`](@ref) is associated with two sets of values:
+
+- The **sample space** of the distribution, which does not depend on the arguments.
+
+- The **support** of the distribution, which may depend on the arguments, and is the set of values that has nonzero probability (or probability density). It may be the entire sample space, or it may be a subset of the sample space.
```

Comment:
AFAIK, the usual definition of "support" is the topological closure of this set -- i.e., this set along with its boundary.  As one possible motivation for including the boundary, note that any neighborhood of a boundary point has positive measure -- this is not true of points outside the support.

Reply:
OK. I this is an instance where if we are super precise, it may become unreadable to many users if all the detail is given. I think that the details can be put in parentheses in some cases, or in separate "mathematical details" sections. This sort of tension between readability and precision shows up everywhere in Gen's documentation. I think that a reasonable approach is to have a "mathematically details" subsection in various sections of documentation that gives all the precise details.

Reply:
That sounds good to me.  However I still think you should say the support of the beta distribution is [0, 1], not (0, 1).  Wdyt?

Reply:
I'm not sure. I just looked up on [Wikipedia](https://en.wikipedia.org/wiki/Beta_distribution) and saw this:

> However, the inclusion of x = 0 and x = 1 does not work for α, β < 1; accordingly, several other authors, including W. Feller,[5][6][7] choose to exclude the ends x = 0 and x = 1, (so that the two ends are not actually part of the domain of the density function) and consider instead 0 < x < 1. 

In general, I would prefer to avoid boundary cases in the definitions of things when we can, to encourage users to avoid writing models that are unecessarily deterministic. For example, when the value sampled from a Beta distribution is used as a probability, it would be nice to know that it won't be exactly 0 or 1. But I chose (0, 1) in this case mainly as a nudge to encourage a way of thinking about models -- how the distinction is implemented in the implementation is complex and involves characterizing behavior infloating points, which is a big discussion of its own.

Reply:
Ah, I didn't realize that some authors say the support of the beta distribution is (0, 1).  Cool, fine with me to keep it that way then.

There is of course a trickiness in trying to make precise the intuitions you're advocating above -- for example, the value of a density function at any particular point can be changed and the result will still be a valid density function for the same distribution.  I agree that it makes sense to intentionally not go into that level of detail in the Gen docs, unless the issue makes a real difference in what is or isn't possible to do in Gen.